compiler/.gitignore

```diff
@@ -2,4 +2,4 @@
 dist
 .cabal-sandbox
 cabal.sandbox.config
-.stack-work/
+.stack-work
```

Comment:
directory but I removed it since .cabal-sandbox is also a directory and didn't have it